### PR TITLE
Add frag tool to measure nbs fragmentation

### PR DESCRIPTION
**Pull request by cmasone-attic**
Before we can defragment NBS stores, we need to understand how
fragmented they are. This tool provides a measure of fragmentation in
which optimal chunk-graph layout implies that ALL children of a given
parent can be read in one storage-layer operation (e.g. disk read, S3
transaction, etc).

### CODE REVIEW

**Comment by cmasone-attic** [comment]
PTAL

File: go/nbs/table_reader.go [hunk 1]
```diff
@@ -345,6 +301,72 @@ func (tr tableReader) getMany(reqs []getRecord) (remaining bool) {
 	return
 }
 
+// findOffsets iterates over |reqs| and |tr.prefixes| (both sorted by
+// address) to build the set of table locations which must be read in order to
+// find each chunk specified by |reqs|. If this table contains all requested
+// chunks remaining will be set to false upon return. If some are not here,
+// then remaining will be true.
+func (tr tableReader) findOffsets(reqs []getRecord) (ors offsetRecSlice, remaining bool) {
```
Comment (cmasone-attic) [inline go/nbs/table_reader.go]: `findOffsets()` and `canReadAhead()` are intended to have the same semantics as the code they were refactored from, so please check me and make sure I didn't change some edge case that's not covered by tests or something

File: go/nbs/store.go [hunk 2]
```diff
@@ -200,6 +200,32 @@ func (nbs *NomsBlockStore) GetMany(hashes []hash.Hash) []chunks.Chunk {
 	return resp
 }
 
+func (nbs *NomsBlockStore) CalcReads(hashes []hash.Hash) (reads int, split bool) {
+	reqs := make([]getRecord, len(hashes))
```
Comment (rafael-atticlabs) [inline go/nbs/store.go]: make a utility for this which is used here & GetMany() to produce this slice of getRecords from hash requests?

**Review by ghost — APPROVED** [review]
LGTM

File: go/nbs/frag/main.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	for i := 0; i < concurrency; i++ {
+		go func() {
+			for ref := range refs {
+				mu.Lock()
+				visited := visitedRefs[ref.TargetHash()]
+				visitedRefs[ref.TargetHash()] = true
+				mu.Unlock()
+
+				if !visited {
+					if ref.Height() >= uint64(2) {
```
Comment (cmasone-attic) [inline go/nbs/frag/main.go]: Why, no! It cannot.

File: go/nbs/store.go (hunk 2 above)
Comment (cmasone-attic) [inline go/nbs/store.go]: done

File: go/nbs/frag/main.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+						numbers <- record{count: 1}
+					}
+
+					if ref.Height() > 1 {
```
Comment (cmasone-attic) [inline go/nbs/frag/main.go]: done
